From 0f727f892d11990e2d7f031ba46078316e61b88c Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Mon, 26 Nov 2018 17:00:26 +0200
Subject: [PATCH 0807/1051] dts: a8040: a7040: enable crypto on board level
 device tree

This patch disable crypto on SoC level and enable in board level
device tree.

Change-Id: I99eaf919d53e9b584dc445453af98f651b4e7658
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/61326
Reviewed-by: Igal Liberman <igall@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-3900-db-vd-A.dts |  4 ++++
 arch/arm64/boot/dts/marvell/armada-3900-db-vd-B.dts |  4 ++++
 arch/arm64/boot/dts/marvell/armada-7040-db-B.dts    |  4 ++++
 arch/arm64/boot/dts/marvell/armada-7040-db-C.dts    |  4 ++++
 arch/arm64/boot/dts/marvell/armada-7040-db-D.dts    |  4 ++++
 arch/arm64/boot/dts/marvell/armada-7040-db-E.dts    |  3 +++
 arch/arm64/boot/dts/marvell/armada-7040-db.dts      |  4 ++++
 arch/arm64/boot/dts/marvell/armada-8040-db.dts      |  8 ++++++++
 arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts   |  8 ++++++++
 arch/arm64/boot/dts/marvell/armada-80x0.dtsi        | 11 -----------
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi       |  1 +
 11 files changed, 44 insertions(+), 11 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-3900-db-vd-A.dts b/arch/arm64/boot/dts/marvell/armada-3900-db-vd-A.dts
index ff69ffbd207f..767456ebf97b 100644
--- a/arch/arm64/boot/dts/marvell/armada-3900-db-vd-A.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3900-db-vd-A.dts
@@ -148,3 +148,7 @@
 	/* Generic PHY, providing serdes lanes */
 	phys = <&cp0_comphy4 1>;
 };
+
+&cp0_crypto {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-3900-db-vd-B.dts b/arch/arm64/boot/dts/marvell/armada-3900-db-vd-B.dts
index c0c78669b969..0d199206c22e 100644
--- a/arch/arm64/boot/dts/marvell/armada-3900-db-vd-B.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3900-db-vd-B.dts
@@ -148,3 +148,7 @@
 	/* Generic PHY, providing serdes lanes */
 	phys = <&cp0_comphy4 1>;
 };
+
+&cp0_crypto {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts
index 709212035a69..8cbdf11a0448 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts
@@ -84,3 +84,7 @@
 	phy = <&cp0_phy3>;
 	phy-mode = "rgmii-id";
 };
+
+&cp0_crypto {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-C.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-C.dts
index f8ed7012c08e..70b958cbb910 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-C.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-C.dts
@@ -107,3 +107,7 @@
 	phy = <&cp0_phy1>;
 	phy-mode = "rgmii-id";
 };
+
+&cp0_crypto {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts
index 00c13b572e7b..c57a1e6193eb 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts
@@ -126,3 +126,7 @@
 	phy = <&cp0_phy1>;
 	phy-mode = "rgmii-id";
 };
+
+&cp0_crypto {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts
index c02e7e039bb5..7c6bed6975c6 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts
@@ -100,3 +100,6 @@
 	phy-mode = "rgmii-id";
 };
 
+&cp0_crypto {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dts b/arch/arm64/boot/dts/marvell/armada-7040-db.dts
index 3480d8337477..3794b642826f 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dts
@@ -153,3 +153,7 @@
 	phy = <&phy1>;
 	phy-mode = "rgmii-id";
 };
+
+&cp0_crypto {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dts b/arch/arm64/boot/dts/marvell/armada-8040-db.dts
index 3dc551796b55..fe9b73c3a6d1 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dts
@@ -214,3 +214,11 @@
 &cp0_sdhci0 {
 	status = "okay";
 };
+
+&cp0_crypto {
+	status = "okay";
+};
+
+&cp1_crypto {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
index 13022d84c6c5..70dad1c7d373 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
@@ -402,3 +402,11 @@
 	/* Generic PHY, providing serdes lanes */
 	phys = <&cp1_comphy2 0>;
 };
+
+&cp0_crypto {
+	status = "okay";
+};
+
+&cp1_crypto {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-80x0.dtsi b/arch/arm64/boot/dts/marvell/armada-80x0.dtsi
index 4af5322c7df8..af14253e2222 100644
--- a/arch/arm64/boot/dts/marvell/armada-80x0.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-80x0.dtsi
@@ -106,14 +106,3 @@
 		};
 	};
 };
-
-&cp1_crypto {
-	/*
-	 * The cryptographic engine found on the cp110
-	 * master is enabled by default at the SoC
-	 * level. Because it is not possible as of now
-	 * to enable two cryptographic engines in
-	 * parallel, disable this one by default.
-	 */
-	status = "disabled";
-};
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index c600d528f340..d2e4387ea494 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -498,6 +498,7 @@
 				 <&CP110_LABEL(clk) 1 17>;
 			cell-index = <CP110_EIP197_INDEX>;
 			dma-coherent;
+			status = "disabled";
 		};
 	};
 
-- 
2.17.1

